wal/wal.go

```diff
@@ -707,25 +706,38 @@ func (r *segmentBufReader) Close() (err error) {
 	return err
 }
 
+// Read implements io.Reader, expect that we occasionally return err=nil
+// when n != len(b), which violates the inteface.  Only use this with
+// io.ReadFull.
```

Comment:
Can you explain why we do that?

Reply:
Wait, it infact doesn't violate the interface. https://golang.org/pkg/io/#Reader It's okay to return early.

Reply:
Yeah fair enough, don't know what I was thinking.